app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  before_action :sign_up_if_not_logged_in
```

Comment:
- [ ] Devise already have similar helper method, instead of implementing your own. [`authenticate_user!`](https://github.com/plataformatec/devise#controller-filters-and-helpers)

Reply:
We had implemented our own to redirect to the registration page instead of devise's which redirects to the login. But we have changed it either way